numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3124,7 +3127,32 @@ finish:
  **                                 ARGFUNC                                 **
  *****************************************************************************
  */
+#if defined( __arm__ ) || defined(__aarch64__)
+    int32_t _mm_movemask_epi8_neon(uint8x16_t input)
+    {
```

Comment:
this code can be optimized better if you used `vpaddlq_*` instead of splitting the vectors into two halves.
look at how OpenCV implement it:
https://github.com/opencv/opencv/blob/515a06cedf354b9c239200298c416db2a213afd6/modules/core/include/opencv2/core/hal/intrin_neon.hpp#L1374-L1380

Reply:
Thanks for your advise, The performance has increased by 6% more.